packages/sanity/src/form/studio/inputResolver/helpers.ts

```diff
@@ -9,15 +9,30 @@ export function getOption(type: SchemaType, optionName: string) {
 const PSEUDO_OBJECTS = ['array', 'file', 'image', 'reference']
 const HIDDEN_FIELDS = ['asset', 'crop', 'hotspot', '_ref', '_weak']
 const NO_LEVEL_LAYOUTS = ['tags']
+const NO_LEVEL_TYPES = ['slug']
 
-export function getObjectFieldLevel(field: ObjectFieldProps): number {
+type PartialObjectSchema = Pick<ObjectSchemaType, 'name' | 'options'> & {
+  type?: PartialObjectSchema
+  fields?: {name: string}[]
+}
+
+export interface PartialObjectFieldProps {
+  schemaType: PartialObjectSchema
+  level: number
+}
+
+export function getObjectFieldLevel(field: PartialObjectFieldProps): number {
```

Comment:
I have this hunch that this information should be encapsulated in the field props/field definition time when you have more local context. This logic is very hard to understand imo. I can investigate if we can move this if you want, otherwise I'll just fix it (as written this is wrong since slug root type is 'object'.